### PR TITLE
[#48] Remove acceptNewFields routes

### LINKED ISSUE #48: Remove acceptNewFields
According to this issue https://github.com/meilisearch/integration-guides/issues/39

### PR DESCRIPTION
**Pull request by Ethenyl**
# Summary

Fixes #48.

Remove `acceptNewFields` routes and add default values for `searchableAttributes` and `displayedAttributes` in `Settings`.

# Miscellaneous

Added UT for testing default values and removed `distinctAttributes` from JSON test payload.

### CODE REVIEW

File: Sources/MeiliSearch/Model/Setting.swift [hunk 1]
```diff
@@ -3,53 +3,42 @@ import Foundation
...
-      stopWords: [String],
-      synonyms: [String: [String]],
-      acceptNewFields: Bool) {
-      self.rankingRules = rankingRules
-      self.searchableAttributes = searchableAttributes
-      self.displayedAttributes = displayedAttributes
-      self.stopWords = stopWords
-      self.synonyms = synonyms
-      self.acceptNewFields = acceptNewFields
+    public init(rankingRules: [String], searchableAttributes: [String], displayedAttributes: [String], stopWords: [String], synonyms: [String: [String]]) {
```
Comment (ppamorim) [inline Sources/MeiliSearch/Model/Setting.swift:26]: Is this init needed since `Setting` is a struct?

Comment (Ethenyl) [reply Sources/MeiliSearch/Model/Setting.swift:26]: Nope it can be completely removed, I will do it right away.

Comment (Ethenyl) [reply Sources/MeiliSearch/Model/Setting.swift]: Removed in `395e888`.

**Review by ppamorim — APPROVED** [review]

**Comment by ppamorim** [comment]
@curquiza Could you please review and merge if approved?

File: Tests/MeiliSearchUnitTests/SettingsTests.swift [hunk 2]
```diff
@@ -17,7 +17,6 @@ class SettingsTests: XCTestCase {
             "exactness",
             "desc(release_date)"
         ],
-        "distinctAttribute": null,
```
Comment (bidoubiwa) [inline Tests/MeiliSearchUnitTests/SettingsTests.swift:20]: Why is this removed ?

File: Tests/MeiliSearchUnitTests/SettingsTests.swift [hunk 3]
```diff
@@ -42,11 +40,21 @@ class SettingsTests: XCTestCase {
...
+        XCTAssertTrue(stubSetting.stopWords.isEmpty)
+        XCTAssertTrue(stubSetting.synonyms.isEmpty)
+    }
+
     func testGetSetting() {
 
         //Prepare the mock server
 
-        let stubSetting: Setting = buildStubSetting()
+        let stubSetting = buildStubSetting(from: json)
```
Comment (bidoubiwa) [inline Tests/MeiliSearchUnitTests/SettingsTests.swift:57]: Why is the typing removed ?

File: Tests/MeiliSearchUnitTests/SettingsTests.swift [hunk 4]
```diff
@@ -42,11 +40,21 @@ class SettingsTests: XCTestCase {
 
     // MARK: Settings
 
+    func testShouldInstantiateFromEmptyData() {
```
Comment (bidoubiwa) [inline Tests/MeiliSearchUnitTests/SettingsTests.swift:43]: Where is this function called ?

File: Sources/MeiliSearch/Model/Setting.swift [hunk 5]
```diff
@@ -3,53 +3,34 @@ import Foundation
...
-      self.stopWords = stopWords
-      self.synonyms = synonyms
-      self.acceptNewFields = acceptNewFields
-    }
-
-    ///Tries to decode the JSON object to Setting object.
+    /// Tries to decode the JSON object to Setting object.
     public init(from decoder: Decoder) throws {
         let values = try? decoder.container(keyedBy: CodingKeys.self)
+
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/Model/Setting.swift:29]: Seems like distinctAttribute is missing here, should a PR be opened to fix this @ppamorim ?

**Review by bidoubiwa — COMMENTED** [review]
Thanks a lot for this contribution! It's a really good work :)

File: Tests/MeiliSearchUnitTests/SettingsTests.swift (hunk 3 above)
Comment (ppamorim) [reply Tests/MeiliSearchUnitTests/SettingsTests.swift:57]: @bidoubiwa Yes, we would like to keep the types explicit.

File: Tests/MeiliSearchUnitTests/SettingsTests.swift (hunk 2 above)
Comment (Ethenyl) [reply Tests/MeiliSearchUnitTests/SettingsTests.swift:20]: I removed since there's no usage of it whatsoever, if you think it is still relevant, I'll revert this change.

File: Tests/MeiliSearchUnitTests/SettingsTests.swift (hunk 3 above)
Comment (Ethenyl) [reply Tests/MeiliSearchUnitTests/SettingsTests.swift:57]: No issue, what is the reason behind this requirement?

I tend to remove any duplication wherever I can, less code = less headaches but I can revert this if needed.

File: Tests/MeiliSearchUnitTests/SettingsTests.swift (hunk 4 above)
Comment (Ethenyl) [reply Tests/MeiliSearchUnitTests/SettingsTests.swift:43]: It is a test, test are automatically detected and added by Xcode, no need to call them manually.

But I see your point, for Linux I need to add the test manually at the end of the file, I'm not used to that, I never perform test for Linux.

File: Tests/MeiliSearchUnitTests/SettingsTests.swift (hunk 3 above)
Comment (ppamorim) [reply Tests/MeiliSearchUnitTests/SettingsTests.swift:57]: This is needed to enforce that types are the same as expected (question yourself: what's the return type of the function `buildStubSetting` without the need of going to the function?), also it helps Xcode to index the code faster, we suffer for constant problems where Xcode is not capable of indexing it faster than a human. :upside_down_face:

File: Tests/MeiliSearchUnitTests/SettingsTests.swift (hunk 4 above)
Comment (bidoubiwa) [reply Tests/MeiliSearchUnitTests/SettingsTests.swift:43]: I did not know that. I'm used to adding them at the end of the file mostly also to have a quick way of knowing which tests are done in each file

File: Tests/MeiliSearchUnitTests/SettingsTests.swift (hunk 2 above)
Comment (bidoubiwa) [reply Tests/MeiliSearchUnitTests/SettingsTests.swift:20]: You are right, it is missing.

File: Tests/MeiliSearchUnitTests/SettingsTests.swift (hunk 3 above)
Comment (Ethenyl) [reply Tests/MeiliSearchUnitTests/SettingsTests.swift:57]: Ok, I never encountered those kind of issues before, even on far bigger projects (which mixed ObjC and Swift) but I will follow your coding guidelines.

As for the naming of the function, it should be called getStubSetting to say that it actually returns this 😉

File: Tests/MeiliSearchUnitTests/SettingsTests.swift (hunk 4 above)
Comment (Ethenyl) [reply Tests/MeiliSearchUnitTests/SettingsTests.swift:43]: Well, I forgot to add, functions prefixed with `test` will be counted as tests 😄

File: Tests/MeiliSearchUnitTests/SettingsTests.swift (hunk 3 above)
Comment (Ethenyl) [reply Tests/MeiliSearchUnitTests/SettingsTests.swift]: Fixed in `0eeac0c`.

**Comment by Ethenyl** [comment]
Just FYI, I had to reword my last commit since the build failed to restart the build.

And without changing anything, the build completed successfully this time, the failing test was `testDeleteDocument` in `DocumentsTests.swift`.

**Comment by ppamorim** [comment]
@Ethenyl Thank you very much, we really appreciate your effort in this PR.

Regarding the failing `testDeleteDocument` test, don't worry, we are sorting these tests in the PR #54.

**Review by ppamorim — APPROVED** [review]

**Review by bidoubiwa — APPROVED** [review]
Thanks a lot

**Review by curquiza — APPROVED** [review]
